webapp/src/ts/modules/analytics/analytics-target-aggregates-sidebar-filter.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { Store } from '@ngrx/store';
+
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
+import { ContactTypesService } from '@mm-services/contact-types.service';
+import { GetDataRecordsService } from '@mm-services/get-data-records.service';
+import { SettingsService } from '@mm-services/settings.service';
+import { TelemetryService } from '@mm-services/telemetry.service';
+import { UserSettingsService } from '@mm-services/user-settings.service';
+
+const FACILITY =  'Facility';
+@Component({
+  selector: 'mm-analytics-target-aggregates-sidebar-filter',
+  templateUrl: './analytics-target-aggregates-sidebar-filter.component.html'
+})
+export class AnalyticsTargetAggregatesSidebarFilterComponent implements OnInit, OnDestroy {
+
+  private globalActions;
+  subscriptions: Subscription = new Subscription();
+  error;
+  isOpen = false;
+  userFacilities;
+  selectedFacilityId;
+  facilityLabel;
+
+  constructor(
+    private store: Store,
+    private contactTypesService: ContactTypesService,
+    private getDataRecordsService: GetDataRecordsService,
+    private settingsService: SettingsService,
+    private telemetryService: TelemetryService,
+    private userSettingsService: UserSettingsService,
+  ) {
+    this.globalActions = new GlobalActions(store);
+  }
+
+  async ngOnInit() {
+    try {
+      this.subscribeToStore();
+      await this.getUserFacility();
+      await this.setFacilityLabel();
+    } catch (err) {
+      this.error = true;
+      console.error('Error initializing Target Sidebar component', err);
+    }
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  private subscribeToStore() {
+    const subscription = this.store.select(Selectors.getSidebarFilter).subscribe(({ isOpen }) => {
+      this.isOpen = isOpen;
+    });
+    this.subscriptions.add(subscription);
+  }
+
+  toggleSidebarFilter() {
+    this.isOpen = !this.isOpen;
+    this.globalActions.setSidebarFilter({ isOpen: this.isOpen });
+
+    if (this.isOpen) {
+      // Counting every time the user opens the sidebar filter in analytics_targets_aggregrate tab.
+      this.telemetryService.record('sidebar_filter:analytics_target_aggregates:open');
+    }
+  }
+
+  private getUserFacility() {
+    return this.userSettingsService
+      .get()
+      .then((userSettings: any) => {
+        return this.getPlaces(userSettings.facility_id);
+      });
+  }
```

Comment:
This is another function that can be moved to `userSettingsService`. We have in other places the same need to get the user facility data; we can centralize it for maintenance 
That includes `private getPlaces(facilityId) { return this.getDataRecordsService` part, so all in 1 function in `userSettingsService`